pkg/designate/pool_update.go

```diff
@@ -0,0 +1,132 @@
+package designate
+
+import (
+	"fmt"
+
+	designatev1beta1 "github.com/openstack-k8s-operators/designate-operator/api/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PoolUpdateCommand      = "/usr/bin/designate-manage pool update"
+	DesignateConfigVolume  = "designate-config"
+	DesignateConfigMount   = "/etc/designate"
+	DesignateConfigKeyPath = "designate.conf"
+	DesignatePoolsYamlPath = "pools.yaml"
+)
+
+func PoolUpdateJob(
+	instance *designatev1beta1.Designate,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.Job {
+	runAsUser := int64(0)
+	volumeMounts := GetVolumeMounts(instance.Name)
+	volumes := GetVolumes(instance.Name)
+
+	volumes = append(volumes, corev1.Volume{
+		Name: "pools-yaml-config",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: PoolsYamlConfigMap,
+				},
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "pools-yaml-content",
+						Path: DesignatePoolsYamlPath,
+					},
+				},
+			},
+		},
+	})
+
+	volumes = append(volumes, corev1.Volume{
+		Name: DesignateConfigVolume,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: "designate-api-config-data",
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "designate.conf",
```

Comment:
Is this part actually needed. IIUC the file designate.conf will be there already.

Reply:
Now I guess that it will, before I execed the pod and didn't find the correct one. So if I have the correct designate.conf there, I guess I could remove that volume mounting.